site/zenodo_rdm/subcommunities/request.py

```diff
@@ -113,8 +124,180 @@ class ZenodoSubCommunityRequest(SubCommunityRequest):
 
     available_actions = {
         "delete": actions.DeleteAction,
-        "create": SubcommunityCreateAction,
         "cancel": actions.CancelAction,
+        # Custom implemented actions
+        "create": SubcommunityCreateAction,
         "accept": SubcommunityAcceptAction,
         "decline": DeclineSubcommunity,
     }
+
+
+class SubcommunityInvitationCreateAction(actions.CreateAction):
+    """Represents an accept action used to accept a subcommunity."""
+
+    def execute(self, identity, uow):
+        """Execute approve action."""
+        parent = self.request.created_by.resolve()
+        if not parent.children.allow:
+            raise ValidationError("Assigned parent is not allowed to be a parent.")
+        super().execute(identity, uow)
```

Comment:
minor: will be merged with the "Submit" action to become "CreateAndSubmit". it also looks like it's not necessary, since it's the same as the CreateAction from invenio-communities for the subcommunity invitation